spec.html

```diff
@@ -21855,8 +21855,12 @@ <h1>Runtime Semantics: Evaluation</h1>
     </emu-clause>
   </emu-clause>
 
-  <emu-clause id="sec-with-statement">
+  <emu-clause id="sec-with-statement" legacy>
     <h1>The `with` Statement</h1>
+    <emu-note>
+      <p>It is recommended that the Legacy `with` statement is not used in new ECMAScript code. Consider alternatives that are permitted in both strict mode code and non-strict code, such as <emu-xref href="#sec-destructuring-assignment"></emu-xref>.</p>
```

Comment:
```suggestion
      <p>Use of the Legacy `with` statement is discouraged in new ECMAScript code. Consider alternatives that are permitted in both strict mode code and non-strict code, such as <emu-xref href="#sec-destructuring-assignment"></emu-xref>.</p>            
```

…because:

* I think *discouraged* is more forceful than “not recommended”.
* I think it would be good for the spec to use consistent language for features that we want to warn developers away from. And I in general believe for that the term “discouraged” is an apt choice, for reasons outlined in https://github.com/mdn/content/discussions/5549#discussioncomment-814689

Reply:
OK, only after writing the above did it dawn on me that _Legacy_ is the term y’all have already landed on to flag these kinds of cases. I anyway still prefer getting “discouraged” in there rather than just “not recommended”. But I guess as long as y’all have already arrived at _Legacy_ as the flag to consistently use for these cases, I don’t feel that strongly about the ancillary wording

Reply:
I'd be happy to strengthen the wording here. For what it's worth, I borrowed the "It is recommended…" style from elsewhere in the spec. Left wholly to my own devices, I'd probably go with something more active (and assertive), such as:

> Do not use the Legacy `with` statement in new ECMAScript code. Consider alternatives….

Reply:
In the editor call today, we decided to adopt @sideshowbarker's suggested wording. I pushed a commit that updates it.